Jetpack Cloud: Add Monitor product lightbox, product data

mattgawarecki opened the pull request:
***UPDATE:** D117122-code is deployed. All relevant instructions have been struck through.*

~~**NOTE: This PR is not fit to merge until a product has been created for Jetpack Monitor in the WordPress.com Store.**~~

~~Depends on D117122-code.~~
Resolves `1204992567518369-as-1204995327932145`.

## Proposed Changes

* Add a section for disclaimer text to `JetpackProductInfo`.
* Add a property `alwaysExpanded` to `JetpackProductInfoSection`, to force a section to always render in its expanded state.
* Add product and feature slugs for Jetpack Monitor.
* Add lightbox verbiage and icons for Jetpack Monitor.

## Testing Instructions

~~***NOTE:** To test before D117122-code is merged, you'll need a WordPress.com sandbox. For prerequisite test instructions, consult the aforementioned PR.*~~

1. In your test environment, visit the Partner Portal's **Issue License** page (i.e., `/partner-portal/issue-license`).
2. Verify you see Monitor in the list of issuable products.
3. Click the "More about Monitor" link to open its lightbox.
4. Verify the lightbox opens, and that all its verbiage and icons are rendered the same as in its design (Gk3oZYJHerhUZ2KVabMAEx-fi-4016-92602).
5. Click the **Select License** button; verify the lightbox closes and the product card's checkbox is now checked.

### Reference screenshots

<img height="370" src="https://github.com/Automattic/wp-calypso/assets/670067/73f0842e-136a-4e2a-ab23-a71d6447bf9e" />

<img height="370" src="https://github.com/Automattic/wp-calypso/assets/670067/c3829e68-612a-45bf-b167-8f0c1049bd14" />

<img height="370" src="https://github.com/Automattic/wp-calypso/assets/670067/2e9b148c-269c-4886-869d-100170da7bc9" />

<img height="370" src="https://github.com/Automattic/wp-calypso/assets/670067/fe11435b-0a71-437c-bda8-da9fd04aa141" />

Commit:
JetpackProductInfoSection: Add prop to leave a section always expanded

Commit:
JetpackProductInfo: Show product disclaimer text

Commit:
calypso-products: Add constants for Jetpack Monitor

Commit:
getProductIcon: Add icons for Jetpack Monitor

Commit:
calypso-products: Add product info text for Jetpack Monitor

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8607527

Thank you @mattgawarecki for including a screenshot in the description! This is really helpful for our translators.

Commit:
calypso-products: Add feature constants for Jetpack Monitor

Commit:
calypso-products: Add products for Jetpack Monitor

Commit:
useProductDescription: Add description for Jetpack Monitor

Commit:
Add lightbox description text for Monitor

Commit:
JetpackProductInfoSection: make 'alwaysExpanded' prop optional

Commit:
JetpackProductInfoSection: make title optional for always-expanded sections

yashwin commented on packages/calypso-products/src/products-list.ts:
```diff
@@ -432,6 +440,40 @@ export const JETPACK_SITE_PRODUCTS_WITH_FEATURES: Record<
 	},
+
+	[ PRODUCT_JETPACK_MONITOR_YEARLY ]: {
+		product_name: translate( 'Monitor' ),
```
As per the design, the title is `Downtime Monitoring.` I might be missing something here.

yashwin reviewed (commented):
I reviewed this since it was marked as `Ready for review`.

This works as expected. Great job on this! 

A few clarifications and issues I found

- The product name in this PR is `Monitor`, whereas it is `Downtime Monitoring` in the design.
- There is a slight alignment issue in the disclaimer section. Is this intentional? 

<img width="317" alt="Screenshot 2023-08-01 at 11 51 42 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/115e4768-36b3-4d85-93d0-68e44bf23fc1">


- The ticket also mentions the modal should open up from the JITM and upgrade paths from the dashboard. Are we planning to wrap that in a separate PR? If yes, could we please create different ticket so that we don't miss it.

Commit:
JetpackProductInfo: Vertically center asterisk in disclaimer

Commit:
Agency Dashboard: show Monitor lightbox on banner/popover CTA click

mattgawarecki commented:
> The product name in this PR is `Monitor`, whereas it is `Downtime Monitoring` in the design.

Ah, yes -- this is intentional, because of this discussion: pedEI5-NG-p2#comment-498. I suspect the designs might've pre-dated this decision.

> There is a slight alignment issue in the disclaimer section. Is this intentional?

Fixed 👍 

> The ticket also mentions the modal should open up from the JITM and upgrade paths from the dashboard. Are we planning to wrap that in a separate PR? If yes, could we please create different ticket so that we don't miss it.

Good catch! I added event handlers to those spots. 👍

yashwin commented:
@mattgawarecki: Thanks for addressing my comments!

I see a couple more upgrade paths where the modal should be opened. 

Enable monitor if not enabled already > Click on the clock icon next to the monitor toggle.

- Hover over the `Upgrade` badge > Click `Explore`

<img width="462" alt="Screenshot 2023-08-03 at 11 02 20 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/934734a7-3c18-41cc-ad46-e1190657a45a">

- Click the `Upgrade ($0.91/m)` link

<img width="448" alt="Screenshot 2023-08-03 at 11 02 37 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/a58c26ad-a7c2-4a59-ab3f-9005b13f399c">

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8607527).

Commit:
Show upgrade modal from feature restriction badge, upgrade links

mattgawarecki commented:
@yashwin Nice -- I made changes to address those two cases. Test one more time for me?

yashwin reviewed (approved):
LGTM 🚀 Thanks for making the changes. Good job on this!

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8607527).
